fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -462,6 +477,21 @@ public CompletableFuture<Void> commitAsync() {
                 return AsyncUtil.DONE;
             }
         });
+        if (config.isReportConflictingKeys()) {
+            // Do this before the close (below).
+            commit = MoreAsyncUtil.composeWhenCompleteAndHandle(commit, (v, ex) -> {
+                if (ex instanceof CompletionException) {
+                    final Throwable cause = ex.getCause();
+                    if (cause instanceof FDBException) {
```

Comment:
Is it guaranteed that the `commit` error is always a `CompletionExcpetion` wrapping an `FDBException`? Or should this use something like `FDBExceptions.getFDBCause()` to extract the causal `FDBException` (if any) from more complicated error hierarchies?

Reply:
Even if it's guaranteed now, using that is more future proof. Fixed.